src/testUtils/TestRig.react.js

```diff
@@ -1,46 +1,80 @@
-import React from 'react'
-
-export var TestRigComponent = React.createClass({
-  contextTypes: {
-    triggerComplete: React.PropTypes.func.isRequired,
-    finish: React.PropTypes.func
-  },
-  getInitialState: function() {
-    return { keyVal : Math.random()};
-  },
-  componentDidUpdate: function(prevProps, prevState) {
-    this.context.finish()
-  },
-  triggerFinalRender: function() {
-    this.setState({keyVal: Math.random()})
-  },
+import React from 'react/addons'
+var TestUtils = React.addons.TestUtils
+
+export class TestRigComponent extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = { keyVal : Math.random(), testComplete: false }
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    if(this.state.testComplete=== true)
+      this.state.expectationCallback()
+  }
+
+  triggerFinalRender() {
+    this.setState({testComplete: true})
+    //this.setState({keyVal: Math.random()}) -- probably don't need this anymore, but holding onto just in case it's useful again
+  }
+
   render() {
     return <div key={this.state.keyval}>{this.props.children}</div>
   }
-})
+
+  setExpectationCallback(expectationCallback) {
+    this.setState({expectationCallback})
+  }
+}
 
 export default class TestRig {
-  constructor(TestComponent, afterRender) {
-    if (TestComponent) this.boltOn(TestComponent, afterRender)
-    if (!afterRender) this.childContext = { finish: ()=>{console.log("No expectations")} }
+  constructor(TestComponent) {
+    if (TestComponent) this.boltOn(TestComponent)
   }
 
-  boltOn(TestComponent, afterRender) {
+  boltOn(TestComponent) {
     this.div = document.createElement('div');
-    var domNode, component
+    this.component = React.render(<TestRigComponent><TestComponent/></TestRigComponent>, this.div)
+    this.domNode = $(React.findDOMNode(this.component))
+  }
 
-    if (afterRender) this.childContext = {finish: afterRender }
-    //TODO use the parent context thign since this is deprecated
-    React.withContext(this.childContext, ()=> {
-      component = React.render(<TestRigComponent ><TestComponent/></TestRigComponent>, this.div)
-    })
-    this.component = component
-    this.domNode = $(component.getDOMNode())
+  boltOff() {
+    React.unmountComponentAtNode(this.div)
+  }
+
+  setExpectationCallback(expectationCallback) {
+    this.component.setState({expectationCallback})
   }
 
   finish() { this.component.triggerFinalRender() }
-}
 
+  clickButton(label) {
+    var element = this.domNode.find("button:contains('" + label + "')")[0]
+    this.clickElement(element);
```

Comment:
@BJK Maybe combine this and the line above it since `element` is only used once?